### PR TITLE
PIP-1017-insert-size

### PR DESCRIPTION
**Pull request by ottojolanki**
Implements insert size metrics calculations.

### CODE REVIEW

File: tests/integration/test_get_insert_size_metrics.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+from pathlib import Path
+
+
+
```
Comment (keenangraham) [inline tests/integration/test_get_insert_size_metrics.py:6]: Remove space.

File: tests/integration/test_get_insert_size_metrics.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+@pytest.mark.workflow(name='test_get_insert_size_metrics')
+def test_insert_size_metrics_match(test_data_dir, workflow_dir, skip_lines_match):
```
Comment (keenangraham) [inline tests/integration/test_get_insert_size_metrics.py:8]: test_insert_size_metrics_match -> test_get_insert_size_metrics_metrics_match

File: wdl/subworkflows/get_insert_size_metrics.wdl [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            "assume_sorted": "true",
+            "validation_stringency": "LENIENT"
+        }
+        File nuclear_bam
+        Resources resources
+    }
+
+    String insert_size_metrics_out = basename(nuclear_bam, ".bam") + ".CollectInsertSizeMetrics.picard"
+    String histogram_pdf_out = insert_size_metrics_out + ".pdf"
+
```
Comment (keenangraham) [inline wdl/subworkflows/get_insert_size_metrics.wdl:22]: Remove space.

File: wdl/tasks/picard.wdl [hunk 4]
```diff
@@ -65,3 +65,34 @@ task mark_duplicates_with_mate_cigar {
...
+        File bam
+        CollectInsertSizeMetricsParams params
+        Resources resources
+        String out = "CollectInsertSizeMetrics.picard"
+        String pdf_out = "CollectInsertSizeMetrics.picard.pdf"
+    }
+
+    command {
+        java -jar $(which picard.jar) CollectInsertSizeMetrics \
+            INPUT=~{bam} \
```
Comment (keenangraham) [inline wdl/tasks/picard.wdl:81]: Wrap these like https://github.com/ENCODE-DCC/dnase-seq-pipeline/blob/7de1358428d85070b9b641265d07e06a06b537fd/wdl/tasks/picard.wdl#L48-L49

File: wdl/tasks/sed.wdl [hunk 5]
```diff
@@ -28,3 +28,32 @@ task remove_trailing_whitespaces {
         disks: resources.disks
     }
 }
+
+
+task remove_first_n_lines {
+    input {
+        File input_file
```
Comment (keenangraham) [inline wdl/tasks/sed.wdl:35]: input_file -> in?

File: wdl/tasks/picard.wdl (hunk 4 above)
Comment (ottojolanki) [reply wdl/tasks/picard.wdl:81]: I on purpose did not. The ones unwrapped are mandatory ones.

File: wdl/tasks/sed.wdl (hunk 5 above)
Comment (ottojolanki) [reply wdl/tasks/sed.wdl:35]: yeah, I suppose it is quite clear from the type.

Comment (ottojolanki) [reply wdl/tasks/sed.wdl]: changed the other task in `sed.wdl` as well.

File: wdl/tasks/picard.wdl (hunk 4 above)
Comment (keenangraham) [reply wdl/tasks/picard.wdl:81]: Should make it consistent with what's already in the file.

**Comment by keenangraham** [comment]
Need to rebase.

File: tests/integration/test_get_insert_size_metrics.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import pytest
+
+from pathlib import Path
+
+
+@pytest.mark.workflow(name='test_get_insert_size_metrics')
```
Comment (keenangraham) [inline tests/integration/test_get_insert_size_metrics.py:6]: The idea is the name of test `test_get_insert_size_metrics` and what it is asserting `metrics_match`. So `test_get_insert_size_metrics_match` -> `test_get_insert_size_metrics_metrics_match`

**Pull request merged**